.github/workflows/auto-update.yml

```diff
@@ -50,21 +50,24 @@ jobs:
           cd graypaper
           echo "VERSION=v$(git describe --tags --abbrev=0)" >> "$GITHUB_OUTPUT"
           cd ..
-      - name: Commit changes to the main branch
-        uses: EndBug/add-and-commit@v9
-        with:
-          default_author: github_actions
-          message: Bump to ${{steps.graypaper_update.outputs.VERSION}}
-          tag: ${{steps.graypaper_update.outputs.VERSION}}
+      - name: Use git as github actions bot
+        run: |
+          BOT_NAME="github-actions[bot]"
+          BOT_EMAIL="$(gh api /users/github-actions[bot] | jq -r '.id')+$BOT_NAME@users.noreply.github.com"
+          git config --global user.name $BOT_NAME
+          git config --global user.email $BOT_EMAIL
+      - name: Create a PR and merge changes to the main branch
+        run: |
+          git checkout -b feature/${{steps.graypaper_update.outputs.VERSION}}
+          git add .
+          git commit -m "Bumped Graypaper to ${{steps.graypaper_update.outputs.VERSION}}"
+          git push --set-upstream origin feature/${{steps.graypaper_update.outputs.VERSION}}
+          gh pr create -B main -H feature/$VERSION \
+            --title "Bump Graypaper to ${{steps.graypaper_update.outputs.VERSION}}" --body ""
+          gh pr merge --merge --delete-branch
```

Comment:
I guess this will create a merge commit. Can we rebase instead?

Reply:
yes we can, of course